src/Orleans/SystemTargetInterfaces/IRemoteGrainDirectory.cs

```diff
@@ -22,7 +22,7 @@ internal interface IGrainInfo
         bool AddActivation(ActivationId act, SiloAddress silo);
         ActivationAddress AddSingleActivation(GrainId grain, ActivationId act, SiloAddress silo, GrainDirectoryEntryStatus registrationStatus);
         bool RemoveActivation(ActivationId act, UnregistrationCause cause, out IActivationInfo entry, out bool wasRemoved);
-        bool Merge(GrainId grain, IGrainInfo other);
+        bool Merge(GrainId grain, IGrainInfo other, IInternalGrainFactory grainFactory);
```

Comment:
Change looks good, thanks. I'll investigate the test failures, which are probably unrelated, and merge